src/image_modifications.cpp

```diff
@@ -761,28 +760,35 @@ REGISTER_MOD_PARSER(CROP_TRANSPARENCY, )
 	return std::make_unique<crop_transparency_modification>();
 }
 
+// TODO: a more general solution?
+bool in_range(int val, int min, int max)
+{
+	return min < val && val < max;
+}
+
 // Black and white
 REGISTER_MOD_PARSER(BW, args)
 {
-	const std::vector<std::string>& params = utils::split(args, ',');
+	const auto params = utils::split_view(args, ',');
 
 	if(params.size() != 1) {
 		ERR_DP << "~BW() requires  exactly one argument";
 		return nullptr;
 	}
 
-	try {
-		int threshold = std::stoi(params[0]);
-		if(threshold < 0 || threshold > 255) {
-			ERR_DP << "~BW() argument out of range 0 - 255";
-			return nullptr;
-		}  else {
-			return std::make_unique<bw_modification>(threshold);
-		}
-	} catch (const std::invalid_argument&) {
+	// TODO: maybe get this directly as uint8_t?
```

Comment:
I think having an error message for an out of range value is kinda important, so if getting it as `uint8_t` renders that more difficult, than you shouldn't.